lib/modules/manager/helmfile/artifacts.spec.ts

```diff
@@ -0,0 +1,180 @@
+import { join } from 'upath';
+import { envMock, mockExecAll } from '../../../../test/exec-util';
+import { Fixtures } from '../../../../test/fixtures';
+import { env, fs, mocked } from '../../../../test/util';
+import { GlobalConfig } from '../../../config/global';
+import type { RepoGlobalConfig } from '../../../config/types';
+import * as docker from '../../../util/exec/docker';
+import * as hostRules from '../../../util/host-rules';
+import * as _datasource from '../../datasource';
+import type { UpdateArtifactsConfig } from '../types';
+import * as helmfile from '.';
+
+jest.mock('../../datasource');
+jest.mock('../../../util/exec/env');
+jest.mock('../../../util/http');
+jest.mock('../../../util/fs');
+jest.mock('../../../util/git');
+
+const datasource = mocked(_datasource);
+
+const adminConfig: RepoGlobalConfig = {
+  localDir: join('/tmp/github/some/repo'), // `join` fixes Windows CI
+  cacheDir: join('/tmp/renovate/cache'),
+  containerbaseDir: join('/tmp/renovate/cache/containerbase'),
+};
+
+const config: UpdateArtifactsConfig = {};
+
+const chartFile = Fixtures.get('helmfile.yaml');
+const lockFile = Fixtures.get('helmfile.lock');
+const lockFileTwo = Fixtures.get('helmfile-two.lock');
+
+describe('modules/manager/helmfile/artifacts', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.resetModules();
+
+    env.getChildProcessEnv.mockReturnValue(envMock.basic);
+    GlobalConfig.set(adminConfig);
+    docker.resetPrefetchedImages();
+    hostRules.clear();
+  });
+
+  afterEach(() => {
+    GlobalConfig.reset();
+  });
+
+  it('returns null if no helmfile.lock found', async () => {
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await helmfile.updateArtifacts({
+        packageFileName: 'helmfile.yaml',
+        updatedDeps,
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+
+  it('returns null if updatedDeps is empty', async () => {
+    expect(
+      await helmfile.updateArtifacts({
+        packageFileName: 'helmfile.yaml',
+        updatedDeps: [],
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+
+  it('returns null if unchanged', async () => {
+    fs.getSiblingFileName.mockReturnValueOnce('helmfile.lock');
+    const execSnapshots = mockExecAll();
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache'
+    );
+    fs.getParentDir.mockReturnValue('');
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await helmfile.updateArtifacts({
+        packageFileName: 'helmfile.yaml',
+        updatedDeps,
+        newPackageFileContent: chartFile,
+        config,
+      })
+    ).toBeNull();
+    expect(execSnapshots).toMatchSnapshot();
```

Comment:
don't use snapshots, check other managers for samples

Reply:
Made an attempt at this in 7584677